src/TesApi.Web/Management/BatchQuotaVerifier.cs

```diff
@@ -106,30 +107,40 @@ public async Task CheckBatchAccountQuotasAsync(VirtualMachineInformation virtual
 
         if (needPoolOrJobQuotaCheck)
         {
-            if (batchUtilization.ActiveJobsCount + 1 > batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota)
-            {
-                throw new AzureBatchQuotaMaxedOutException($"No remaining active jobs quota available. There are {batchUtilization.ActivePoolsCount} active jobs out of {batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota}.");
-            }
+            var (_, exception) = CheckBatchAccountPoolOrJobQuotasImpl(required: 1, batchUtilization.ActiveJobsCount, batchUtilization.ActivePoolsCount, batchVmFamilyBatchQuotas.ActiveJobAndJobScheduleQuota, batchVmFamilyBatchQuotas.PoolQuota);
 
-            if (batchUtilization.ActivePoolsCount + 1 > batchVmFamilyBatchQuotas.PoolQuota)
+            if (exception is not null)
             {
-                throw new AzureBatchQuotaMaxedOutException($"No remaining pool quota available. There are {batchUtilization.ActivePoolsCount} pools in use out of {batchVmFamilyBatchQuotas.PoolQuota}.");
+                throw exception;
             }
         }
 
-        if (needCoresUtilizationQuotaCheck)
+        if ((batchUtilization.TotalCoresInUse + workflowCoresRequirement) > batchVmFamilyBatchQuotas.TotalCoreQuota)
         {
-            if ((batchUtilization.TotalCoresInUse + workflowCoresRequirement) > batchVmFamilyBatchQuotas.TotalCoreQuota)
-            {
-                throw new AzureBatchQuotaMaxedOutException($"Not enough core quota remaining to schedule task requiring {workflowCoresRequirement} {(isDedicated ? "dedicated" : "low priority")} cores. There are {batchUtilization.TotalCoresInUse} cores in use out of {batchVmFamilyBatchQuotas.TotalCoreQuota}.");
-            }
+            throw new AzureBatchQuotaMaxedOutException($"Not enough core quota remaining to schedule task requiring {workflowCoresRequirement} {(isDedicated ? "dedicated" : "low priority")} cores. There are {batchUtilization.TotalCoresInUse} cores in use out of {batchVmFamilyBatchQuotas.TotalCoreQuota}.");
+        }
+    }
 
-            if (isDedicatedAndPerVmFamilyCoreQuotaEnforced && batchUtilization.DedicatedCoresInUseInRequestedVmFamily + workflowCoresRequirement > batchVmFamilyBatchQuotas.VmFamilyQuota)
-            {
+    /// <inheritdoc cref="IBatchQuotaProvider"/>
+    public async Task<(int exceeded, Exception exception)> CheckBatchAccountPoolOrJobQuotasAsync(int required, CancellationToken cancellationToken)
```

Comment:
It's used in `BatchScheduler` to extend the quota checks to cover the case where multiple pools are being added in one call from the Scheduler, since we now add tasks in parallel in groups instead of serially and the first round of quota checks assume that only one pool/job are added at a time.

Operationally, two kinds of values are returned: exceeded > 0 and exception is not null, where exceeded is the number of new pools that cannot be created because of quota, or exceeded == 0 and exception is null, where all the requested pools can be created.